### PR TITLE
add vehicle wheel status info to vehicle status

### PR DESCRIPTION
**Pull request by ChrisFajardo-TRI**
velocity, height, pitch_angle, steer_angle

### CODE REVIEW

**Review by andrewbest-tri — APPROVED** [review]
LGTM

**Comment by ChrisFajardo-TRI** [comment]
need to wait for the carla_msg submodule PR to merge first https://github.com/ToyotaResearchInstitute/ros-carla-msgs/pull/2

File: carla_ros_bridge/src/carla_ros_bridge/ego_vehicle.py [hunk 1]
```diff
@@ -127,6 +129,16 @@ def send_vehicle_msgs(self, frame, timestamp):
         vehicle_status.control.reverse = self.carla_actor.get_control().reverse
         vehicle_status.control.gear = self.carla_actor.get_control().gear
         vehicle_status.control.manual_gear_shift = self.carla_actor.get_control().manual_gear_shift
+        for wheel_idx in range(4):
+            wheel_location = VehicleWheelLocation(wheel_idx)
+            wheel_status = CarlaVehicleWheelStatus()
+            wheel_status.speed = self.carla_actor.get_wheel_speed(wheel_location) * 30.0 / math.pi
```
Comment (andrewbest-tri) [inline carla_ros_bridge/src/carla_ros_bridge/ego_vehicle.py]: why 30 here? Wheel radius?  linear speed = angular speed * pi * diameter, so angular speed should be linear speed / (diameter*pi) right?

File: carla_ros_bridge/src/carla_ros_bridge/ego_vehicle.py [hunk 2]
```diff
@@ -127,6 +129,16 @@ def send_vehicle_msgs(self, frame, timestamp):
...
+            wheel_status.steer_angle = self.carla_actor.get_wheel_steer_angle(wheel_location)
+            wheel_status.pitch_angle = self.carla_actor.get_wheel_pitch_angle(wheel_location)
+            wheel_status.height = self.carla_actor.get_wheel_height(wheel_location)
+            wheel_status.lat_slip = self.carla_actor.get_wheel_lat_slip(wheel_location) * 180 / math.pi
```
Comment (andrewbest-tri) [inline carla_ros_bridge/src/carla_ros_bridge/ego_vehicle.py]: what are the units on these? This one has a rad to degree conversion but not the one below?

**Review by andrewbest-tri — CHANGES_REQUESTED** [review]
please  comment on conversions. I am not sure I understand them.

File: carla_ros_bridge/src/carla_ros_bridge/ego_vehicle.py (hunk 1 above)
Comment (ChrisFajardo-TRI) [reply carla_ros_bridge/src/carla_ros_bridge/ego_vehicle.py]: accident left these conversions in when debugging. speed will be deg/sec as commented in the msg definition

File: carla_ros_bridge/src/carla_ros_bridge/ego_vehicle.py (hunk 2 above)
Comment (ChrisFajardo-TRI) [reply carla_ros_bridge/src/carla_ros_bridge/ego_vehicle.py]: accident left these conversions in when debugging. all angles are degrees as commented in the new msg definition.

**Review by andrewbest-tri — APPROVED** [review]
LGTM

**Comment by ChrisFajardo-TRI** [comment]
will close in favor of going to anrp ros bridge directly(https://github.com/anrp-tri/ros-bridge/pull/8)

**Pull request closed without merging**